Basic_obstacle_avoidance/Basic_obstacle_avoidance.ino

```diff
@@ -0,0 +1,94 @@
+#include <Smartcar.h>
+
+
+const int TRIGGER_PIN           = 6; // D6
+const int ECHO_PIN              = 7; // D7
+const unsigned int MAX_DISTANCE = 1000;
+const auto pulsesPerMeter = 600;
+int initialSensorDistance = 0;
+int laterSensorDistance =0;
+
+ArduinoRuntime arduinoRuntime;
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins); 
+DirectionlessOdometer leftOdometer{
+    arduinoRuntime,
+    smartcarlib::pins::v2::leftOdometerPin,
+    []() { leftOdometer.update(); },
+    pulsesPerMeter};
+DirectionlessOdometer rightOdometer{
+    arduinoRuntime,
+    smartcarlib::pins::v2::rightOdometerPin,
+    []() {rightOdometer.update(); },
+    pulsesPerMeter};
+DifferentialControl control(leftMotor, rightMotor); 
+DistanceCar car(arduinoRuntime, control, leftOdometer, rightOdometer);
+SR04 front(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+                
+
+
+void setup()
+{
+    Serial.begin(9600);
+     
+    
+}
+void handleInput()
+{
+    // handle serial input if there is any
+    if (Serial.available())
+    {
+        String input = Serial.readStringUntil('\n');
+        if (input.startsWith("m"))
+        {
+            int throttle = input.substring(1).toInt();
+            car.setSpeed(throttle);
+        }
+        else if (input.startsWith("t"))
+        {
+            int deg = input.substring(1).toInt();
+            car.setAngle(deg);
+        }
+        else if (input.startsWith("c"))
+        {
+          car.enableCruiseControl();
+          int cruiseSpeed = input.substring(1).toInt();
+          car.setSpeed(cruiseSpeed);
+        }
+         else if (input.startsWith("x"))
+         {
+          car.disableCruiseControl();
+          }
+        
+         
+    }
+}
+ 
+void loop()
+{
+    
+    Serial.print("Distance with front obstacle:");
+    Serial.println(front.getDistance());
+    delay(200);
+    car.update(); // Maintain the speed
+    if(front.getDistance()<=180&& movingSituation() && front.getDistance()>0)
```

Comment:
You get calling `getDistance` twice which will result in two potentially **different** values being returned.
For example, what happens if the first time `0` is return and the second time `200`?

Reply:
We are still trying to learn how to integrate with CI hopefully we can sort this issue quickly in the coming days.

Reply:
Where are you getting stuck in regards to the integration with CI?!

Reply:
Also, how does this relate to the problem  that was pointed out here?

Reply:
As you mentioned why the sketches are not build on CI we are still confused with that part, we just tried to put through whatever we could do with obstacle avoidance.

Reply:
As for the code I have stored the getDistance into a new variable named FrontDistance to avoid the things you mentioned. And When I try to push the merged master branch The git gives me this and I do not know how to fix it.

- remote: error: GH006: Protected branch update failed for refs/heads/master.
remote: error: Required status check "build-sketch" is expected. At least 1 approving review is required by reviewers with write access.
To https://github.com/DIT112-V21/group-13.git
 ! [remote rejected] master -> master (protected branch hook declined)
error: failed to push some refs to 'https://github.com/DIT112-V21/group-13.git'

Reply:
Need to create a new pull request

Reply:
> Need to create a new pull request

No, that's not necessary. Update this pull request please.



> As for the code I have stored the getDistance into a new variable named FrontDistance to avoid the things you mentioned. And When I try to push the merged master branch The git gives me this and I do not know how to fix it.
> 
> * remote: error: GH006: Protected branch update failed for refs/heads/master.
>   remote: error: Required status check "build-sketch" is expected. At least 1 approving review is required by reviewers with write access.
>   To https://github.com/DIT112-V21/group-13.git
>   ! [remote rejected] master -> master (protected branch hook declined)
>   error: failed to push some refs to 'https://github.com/DIT112-V21/group-13.git'

Your solution is correct, however we have already covered the git workflow you are supposed to be using. Push to the `Basic_obstacle_avoidance` branch instead. You cannot push directly to `master`.



> As you mentioned why the sketches are not build on CI we are still confused with that part, we just tried to put through whatever we could do with obstacle avoidance.

In the `.github/workflows` directory you can find the configuration that triggers the CI job that builds the sketch. Have a look at it, understand how it works and at the [last line](https://github.com/DIT112-V21/group-13/blob/master/.github/workflows/arduino-build.yml#L25) you will see how the sketch is compiled.
You have two choices, either change what is compiled on CI (and remove whatever you don't need) or put the code you want in the sketch that is already being compiled on CI.